corehq/apps/sso/views/enterprise_admin.py

```diff
@@ -109,7 +115,9 @@ def edit_enterprise_idp_form(self):
             return form_class(
                 self.identity_provider, self.request.POST, self.request.FILES
             )
-        return form_class(self.identity_provider)
+
+        api_fields_enabled = self.show_api_fields()
+        return form_class(self.identity_provider, allow_multi_view_api_keys=api_fields_enabled)
```

Comment:
I think this value needs to be passed into `form_class` above.

Also, curious as to the logic behind all the different names here? `api_fields_enabled`, `show_api_fields`, `allow_multi_view_api_keys` and why `show_api_fields`. Why are all these things named separately? And why was `show_api_fields()` not an `@property`?

Also, `show_api_fields` seems vague for this boolean. I feel that it could get confused with the api secrets / tokens we need for accessing the user-management APIs for each Identity Provider. I have a preference for `show_mutli_view_api_keys_fields`, but I also can see an argument for that being a bit too wordy.

Reply:
Thanks for pointing out the inconsistencies in variable naming here. I've rearranged the code to avoid the property entirely and renamed the variables to all be `uses_api_key_management`, as that is the new section that is being introduced to the form for all of these properties. I realized I didn't like `show_*` because show is often read as a verb, which implies a function.

Both of the issues should be addressed in 8b43ece8ae4a67e73af1861bf8d9b53bc0982263